airbyte-cdk/python/airbyte_cdk/sources/streams/http/auth/oauth.py

```diff
@@ -26,14 +26,16 @@ def __init__(
         client_secret: str,
         refresh_token: str,
         scopes: List[str] = None,
-        refresh_access_token_headers: Mapping[str, Any] = None,
+        refresh_access_token_headers: Optional[Mapping[str, Any]] = None,
+        refresh_access_token_authenticator: Optional[HttpAuthenticator] = None,
```

Comment:
optional authenticator allows us to declare an authenticator instead of directly specify the headers, which can be difficult to express because of the base64 encoding.
Using this pattern, the pinterest authenticator can be expressed as

```
authenticator:
  class_name: "airbyte_cdk.sources.streams.http.auth.oauth.Oauth2Authenticator"
  client_id: "{{ config.client_id }}"
  client_secret: "{{ config.client_secret }}"
	refresh_token: "{{ config.refresh_token }}"
  refresh_access_token_authenticator:
    type: BasicAuthentication
    username: "{{ config.client_id }}"
    password: "{{ config.client_secret }}"
```